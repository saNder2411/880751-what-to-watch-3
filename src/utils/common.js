const TimeShiftInMin = {
  YEAR: 525600,
  MONTH: 43800,
  WEEK: 10080,
  DAY: 1440,
  HOUR: 60,
  MINUTES: 1,
};

const Month = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

const MAX_RATING = 10;

export default class Common {
  static getRandomNumberFromPeriod(max, min = 0) {
    return min + Math.floor((max - min) * Math.random());
  }

  static getRandomDate(shiftInMin = TimeShiftInMin.YEAR * 20, sign = -1) {
    const currentDate = new Date();
    const diffValue = sign * this.getRandomNumberFromPeriod(shiftInMin);

    currentDate.setMinutes(currentDate.getMinutes() + diffValue);

    return +currentDate;
  }

  static getRandomRating() {
    const randomNumber = Common.getRandomNumberFromPeriod(MAX_RATING + 1, 1);
    const randomDecimal = Common.getRandomNumberFromPeriod(MAX_RATING);

    return randomNumber < MAX_RATING ? `${randomNumber}.${randomDecimal}` : `${randomNumber}`;
  }

  static calcCardLevel(ratingVal, [minVal, middleLowVal, middleVal, maxVal]) {
    const ratingToNum = +ratingVal;
    let level = `Bad`;

    if (ratingToNum >= minVal && ratingToNum <= middleLowVal) {
      level = `Normal`;
    } else if (ratingToNum >= middleLowVal && ratingToNum <= middleVal) {
      level = `Good`;
    } else if (ratingToNum >= middleVal && ratingToNum < maxVal) {
      level = `Very good`;
    } else if (ratingToNum >= maxVal) {
      level = `Awesome`;
    }

    return level;
  }

  static parseDateToStr(date) {
    const day = new Date(date).getDate();
    const month = new Date(date).getMonth();
    const year = new Date(date).getFullYear();

    return `${Month[month]} ${day}, ${year}`;
  }

  static dividedArrayInHalf(array) {
    const firstPartLength = ((array.length % 2) + array.length) / 2;

    return [array.slice(0, firstPartLength), array.slice(firstPartLength)];
  }

  static extend(a, b) {
    return Object.assign({}, a, b);
  }
}
