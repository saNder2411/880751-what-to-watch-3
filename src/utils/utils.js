const Month = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];
const MINUTES_IN_HOUR = 60;

const calcCardLevel = (ratingVal, [minVal, middleLowVal, middleVal, maxVal]) => {
  const ratingToNum = +ratingVal;
  let level = `Bad`;

  if (ratingToNum >= minVal && ratingToNum <= middleLowVal) {
    level = `Normal`;
  } else if (ratingToNum >= middleLowVal && ratingToNum <= middleVal) {
    level = `Good`;
  } else if (ratingToNum >= middleVal && ratingToNum < maxVal) {
    level = `Very good`;
  } else if (ratingToNum >= maxVal) {
    level = `Awesome`;
  }

  return level;
};

const dividedArrayInHalf = (array) => {
  const firstPartLength = ((array.length % 2) + array.length) / 2;

  return [array.slice(0, firstPartLength), array.slice(firstPartLength)];
};

const parseDateToStr = (date) => {
  const day = new Date(date).getDate();
  const month = new Date(date).getMonth();
  const year = new Date(date).getFullYear();

  return `${Month[month]} ${day}, ${year}`;
};

const getTimeInHoursAndMinutes = (timeInMin) => {
  const runtime = {
    hours: `0`,
    minutes: `0`,
  };

  if (timeInMin < MINUTES_IN_HOUR) {
    runtime.minutes = (timeInMin < 10) ? `0${timeInMin}` : `${timeInMin}`;
    return runtime.minutes;
  }

  runtime.minutes = timeInMin % MINUTES_IN_HOUR;
  runtime.hours = (timeInMin - runtime.minutes) / MINUTES_IN_HOUR;

  return (runtime.hours) ? `${runtime.hours}h ${runtime.minutes}m` : `${runtime.minutes}m`;
};

const extend = (a, ...b) => Object.assign({}, a, ...b);

export {calcCardLevel, parseDateToStr, dividedArrayInHalf, getTimeInHoursAndMinutes, extend};
