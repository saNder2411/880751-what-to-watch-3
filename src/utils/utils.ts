const Month = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];
const MINUTES_IN_HOUR = 60;

const calcCardLevel = (ratingVal: number, [minVal, middleLowVal, middleVal, maxVal]: Array<number>) => {
  let level = `Bad`;

  if (ratingVal >= minVal && ratingVal <= middleLowVal) {
    level = `Normal`;
  } else if (ratingVal >= middleLowVal && ratingVal <= middleVal) {
    level = `Good`;
  } else if (ratingVal >= middleVal && ratingVal < maxVal) {
    level = `Very good`;
  } else if (ratingVal >= maxVal) {
    level = `Awesome`;
  }

  return level;
};

const dividedArrayInHalf = (array) => {
  const firstPartLength = ((array.length % 2) + array.length) / 2;

  return [array.slice(0, firstPartLength), array.slice(firstPartLength)];
};

const parseDateToStr = (date) => {
  const day = new Date(date).getDate();
  const month = new Date(date).getMonth();
  const year = new Date(date).getFullYear();

  return `${Month[month]} ${day}, ${year}`;
};

const getTimeInHoursAndMinutes = (timeInMin: number) => {
  const runtime = {
    hours: `0`,
    minutes: `0`,
  };

  if (timeInMin < MINUTES_IN_HOUR) {
    runtime.minutes = (timeInMin < 10) ? `0${timeInMin}` : `${timeInMin}`;
    return runtime.minutes;
  }

  runtime.minutes = (timeInMin % MINUTES_IN_HOUR).toString();
  runtime.hours = ((timeInMin - (+runtime.minutes)) / MINUTES_IN_HOUR).toString();

  return (runtime.hours) ? `${runtime.hours}h ${runtime.minutes}m` : `${runtime.minutes}m`;
};

const extend = (a, ...b) => Object.assign({}, a, ...b);

const getAppRoute = (id = `:id`) => ({
  ROOT: `/`,
  CARDS: `/cards/${id}`,
  PLAYER: `/player/${id}`,
  LOGIN: `/login`,
  REVIEW: `/cards/${id}/review`,
  USER_LIST: `/userList`,
});

const noop = () => {};

export {calcCardLevel, parseDateToStr, dividedArrayInHalf, getTimeInHoursAndMinutes,
  extend, getAppRoute, noop};
