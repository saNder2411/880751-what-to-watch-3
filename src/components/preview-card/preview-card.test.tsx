import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {noop} from '../../utils/utils';
import PreviewCard from './preview-card';


const previewCardData = {
  id: 1,
  title: `Bohemian Rhapsody`,
  previewImage: `bohemian-rhapsody`,
  previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

const previewCardHandlers = [noop, noop, noop];

const renderPlayer = () => [];

describe(`Render PreviewCard`, () => {
  it(`Should PreviewCard render correctly without video`, () => {
    const markup = renderer.create(
        <PreviewCard
          isPlaying={false}
          previewCardData={previewCardData}
          previewCardHandlers={previewCardHandlers}
          renderPlayer={renderPlayer}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should PreviewCard render correctly with video`, () => {
    const markup = renderer.create(
        <PreviewCard
          isPlaying={true}
          previewCardData={previewCardData}
          previewCardHandlers={previewCardHandlers}
          renderPlayer={renderPlayer}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
