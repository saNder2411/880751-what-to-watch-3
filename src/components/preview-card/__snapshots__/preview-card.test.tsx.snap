// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PreviewCard Should PreviewCard render correctly with video 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id="1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Bohemian Rhapsody"
      height="175"
      src="bohemian-rhapsody"
      style={
        Object {
          "display": "none",
        }
      }
      width="280"
    />
    <div
      style={
        Object {
          "display": "",
        }
      }
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;

exports[`Render PreviewCard Should PreviewCard render correctly without video 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id="1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Bohemian Rhapsody"
      height="175"
      src="bohemian-rhapsody"
      style={
        Object {
          "display": "",
        }
      }
      width="280"
    />
    <div
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
